File: programme/pattern-2/pattern-2.cpp
// Write a programme to print the given pattern
#include <iostream>
using namespace std;

void pattern(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = n - 1; j > i; j--)
        {
            cout << " ";
        }
        for (int k = 0; k <= i; k++)
        {
            cout << "* ";
        }
        cout << endl;
    }
    for (int i = 1; i < n; i++)
    {
        for (int k = 1; k <= i; k++)
        {
            cout << " ";
        }
        for (int j = n - 1; j > i - 1; j--)
        {
            cout << "* ";
        }
        cout << endl;
    }
}

int main()
{
    int a;
    cout << "Input : ";
    cin >> a;
    pattern(a);
    return 0;
}

File: programme/pattern-2/pattern-2.cs
// Write a programme to print the given pattern

using System;

public class Patern2
{
    public static void Main(string[] args)
    {
        Console.Write("Input  : ");
        if (int.TryParse(Console.ReadLine(), out int num))
        {
            for (int i = 0; i < num*2; i++)
            {
                int stars = i <= num ? i : num * 2 - i;
                string line = "";

                for (int offset = num - stars; offset > 0; offset--) line += " ";
                for (; stars > 0; stars--) line += "* ";

                Console.WriteLine(line);
            }
        }
        else
        {
            Console.WriteLine("Input Err");
        }
    }
}